diffrax/_term.py

```diff
@@ -776,3 +785,135 @@ def _to_vjp(_y, _diff_args, _diff_term):
+
+    $d \mathbf{x}_t = \mathbf{v}_t \, dt$
+
+    $d \mathbf{v}_t = - \gamma \, \mathbf{v}_t \, dt - u \,
+    \nabla \! f( \mathbf{x}_t ) \, dt + \sqrt{2 \gamma u} \, d W_t.$
+    """
+
+    gamma: LangevinX
+    u: LangevinX
+    control: AbstractBrownianPath
+
+    def __init__(self, gamma, u, control: AbstractBrownianPath):
+        self.gamma = gamma
+        self.u = u
+        self.control = control
+
+    def vf(self, t: RealScalarLike, y: LangevinTuple, args: Args) -> LangevinX:
+        x, v = y
+
+        def _fun(_gamma, _u, _v):
+            return jnp.sqrt(2 * _gamma * _u) * jnp.ones(_v.shape, _v.dtype)
+
+        d_v = jtu.tree_map(_fun, self.gamma, self.u, v)
+        return d_v
+
+    def contr(self, t0: RealScalarLike, t1: RealScalarLike, **kwargs) -> _LangevinBM:
+        return self.control.evaluate(t0, t1, **kwargs)
+
+    def prod(self, vf: LangevinX, control: LangevinX) -> LangevinTuple:
+        if isinstance(control, AbstractBrownianIncrement):
+            control = control.W
+
+        dv = vf
+        dw = control
+        v_out = jtu.tree_map(operator.mul, dv, dw)
+        x_out = jtu.tree_map(jnp.zeros_like, v_out)
+        return x_out, v_out
+
+
+_LangevinArgs = tuple[LangevinX, LangevinX, Callable[[LangevinX], LangevinX]]
+
+
+def _langevin_drift(t, y: LangevinTuple, args: _LangevinArgs) -> LangevinTuple:
+    gamma, u, grad_f = args
+    x, v = y
+    f_x = grad_f(x)
+    d_x = v
+    d_v = jtu.tree_map(
+        lambda _gamma, _u, _v, _f_x: -_gamma * _v - _u * _f_x,
+        gamma,
+        u,
+        v,
+        f_x,
+    )
+    d_y = (d_x, d_v)
+    return d_y
+
+
+def _broadcast_pytree(source, target_tree_shape):
+    # Requires that source is a prefix tree of target_tree_shape
+    def _inner_broadcast(_src_arr, _inner_tree_shape):
+        _arr = jnp.asarray(_src_arr)
+        return jtu.tree_map(
+            lambda _leaf: jnp.broadcast_to(_arr, _leaf.shape), _inner_tree_shape
+        )
+
+    return jtu.tree_map(_inner_broadcast, source, target_tree_shape)
+
+
+class LangevinTerm(AbstractTerm):
+    r"""Used to represent the Langevin SDE, given by:
+
+    $d \mathbf{x}_t = \mathbf{v}_t \, dt$
+
+    $d \mathbf{v}_t = - \gamma \, \mathbf{v}_t \, dt - u \,
+    \nabla \! f( \mathbf{x}_t ) \, dt + \sqrt{2 \gamma u} \, d W_t.$
+
+    where $\mathbf{x}_t, \mathbf{v}_t \in \mathbb{R}^d$ represent the position
+    and velocity, $W$ is a Brownian motion in $\mathbb{R}^d$,
+    $f: \mathbb{R}^d \rightarrow \mathbb{R}$ is a potential function, and
+    $ \gamma,\, u\in\mathbb{R}^{d\times d}$ are diagonal matrices representing
+    friction and a dampening parameter.
+    """
+
+    args: _LangevinArgs = eqx.field(static=True)
+    term: MultiTerm[tuple[ODETerm, _LangevinDiffusionTerm]]
+
+    def __init__(self, args, bm: AbstractBrownianPath, x0: LangevinX):
+        r"""**Arguments:**
+
+        - `args`: a tuple of the form $(\gamma, \, u, \, \nabla \! f)$
```

Comment:
Why are these together in `args`, rather than separate arguments?

Reply:
Also, I think spacing commands like `\!` should be avoided in LaTeX like this, as (a) it's often read directly, and (b) many renderers don't support them.

Reply:
> Why are these together in `args`, rather than separate arguments?

Because it makes it easier to pass them on that way. I swapped to have them separate at some point, but I think it looked more awkward.

> Also, I think spacing commands like `\!` should be avoided in LaTeX like this, as (a) it's often read directly, and (b) many renderers don't support them.

Fair enough, I'll get rid of it.